src/libsystemd/sd-bus/bus-message.c

```diff
@@ -1394,8 +1395,10 @@ static void *message_extend_body(
                         }
 
                         /* Readjust pointers */
-                        for (c = m->containers; c < m->containers + m->n_containers; c++)
-                                c->array_size = adjust_pointer(c->array_size, op, os, part->data);
+                        if (m->n_containers > 0) {
```

Comment:
You removed the parens up there, so maybe don't add them here?
And reduce the scope of `c` as above?